pkg/workflow/argo.go

```diff
@@ -239,17 +353,17 @@ func (a *argo) generateAppGroupTpls(ctx context.Context, g *v1alpha1.Application
 	if err != nil {
 		return fmt.Errorf("failed to generate application DAG templates : %w", err)
 	}
-	a.updateWorkflowTemplates(adt...)
+	updateWorkflowTemplates(a.wf, adt...)
```

Comment:
Same comment as before, why this change?

Reply:
@davutkarabay can answer that.

Reply:
Previously the function updated the forward workflow with the templates. Now we are using it to update either the forward workflow or the reverse workflow which is passed as parameter.

Reply:
Isn't the end result still the same or am I missing something?

Reply:
In the previous iteration, we only had the forward workflow (`a.wf`) member in the Argo type struct. So it made sense to make this a method and not a standalone function. But since we have added the reverse workflow as a member of Argo struct, it made sense to make this a function instead of an associate method. This is because we leverage this function both when building the forward workflow and also the reverse workflow

Reply:
@jonathan-innis do you see the usage of updateWorkflowTemplates() in lines #326 and #328 in this file (end of generateReverseWorkflow()). Does this clarify?

Reply:
ah, I see one letter and I missed it, all good then 🚀